tests/integ/modin/index/test_df_series_creation_with_index.py

```diff
@@ -342,31 +344,36 @@ def test_create_with_index_as_data_and_series_as_index(
                 name="mixed series as index",
             ),
             native_pd.Index(["B", 0, None, 3.14]),
+            True,
         ),  # rearranged index values
         (
             native_pd.Series(["A", "B", "C", "D", "E"], name="series"),
             native_pd.Index([3, 4], name="index"),
+            True,
         ),  # subset of index values
         (
             native_pd.Series(
                 list(range(20)), index=native_pd.Index(list(range(20)), name=20)
             ),
             native_pd.Index(list(range(20))),
+            True,
         ),  # all index values match
         (
             native_pd.Series(["A", "V", "D", "R"]),
             native_pd.Index([10, 20, 30, 40], name="none"),
+            True,
         ),  # no index values match
         (
             native_pd.Series([], name="empty series", dtype="int64"),
             native_pd.Index([], name="empty index", dtype="int64"),
+            False,
```

Comment:
Why are we not checking index type here?

Reply:
After an empty index is converted to variant, the inferred_type is now `empty` instead of `integer`. We are currently not ensuring dtype, so would not returning the same index type in this case be a blocker?

Reply:
Hmm - was the index type the same before? It would be nice to have the index type be correct, but I don't think it should be a blocker - maybe add it as a low/medium priority ticket as follow up work?

Reply:
Good point! I created this ticket to keep track of this https://snowflakecomputing.atlassian.net/browse/SNOW-1740545